Review Importing & Exporting Objects how-to (#4568)
mendix/docs: content/en/docs/refguide/modeling/import-and-export/_index.md

---
title: "Importing and Exporting Apps, Modules, Widgets, and Documents"
url: /refguide/import-and-export/
category: "App Modeling"
weight: 12
tags: ["mpk", "import", "export", "document", "module", "widget", "app package"]
---

## 1 Introduction

> In Mendix, you can import various elements others have created (apps, solutions, modules, etc.) and export elements for others to use. The Marketplace provides this functionality on a macro level. You can also use built-in import and export functions to share content between your own apps, import unofficial modules, and share entire apps.

Reviewer: Not sure what customers will get from "on a macro level".
Is there a better way to describe this?
"The Market place provides a platform to share components among the whole Mendix community"?

> There are *.mpk* packages that can contain apps, modules, or widgets; *.mxmodule* packages with add-ons and *.mxsolution* packages solutions. 

Reviewer: ```suggestion
There are *.mpk* packages that can contain apps, modules, or widgets, *.mxmodule* packages containing add-ons, and *.mxsolution* packages containing solutions. 
```

{{% alert color="warning" %}}

> When you have an app package, you cannot import individual modules or widgets separately from it. 

Reviewer: ```suggestion
When you have an app package, you cannot import individual modules or widgets from within it. 
```

{{% /alert %}}

You can import and export the following:
* apps
* modules
* widgets
* documents such as pages or microflows

## 2 Importing Various Elements

{{% alert color="warning" %}}
Be careful when importing elements, as you can overwrite your existing work. Pay attention to prompts and warnings along the way!
{{% /alert %}}

### 2.1 Importing App and Solution Packages {#import-app-package}

A Mendix package (*.mpk*) file can store a complete Mendix app. 

> You can also import a solution package (*.mxsolution*), which is also a complete Mendix app that was developed as out-of-the-box solution.  

Reviewer: ```suggestion
You can also import a solution package (*.mxsolution*), which is a complete Mendix app that was developed as out-of-the-box solution.  
```

> To import a complete an app or solution package, follow these steps:

Reviewer: ```suggestion
To import a complete app or solution package, follow these steps:
```

> 1. Click the **File** menu in and select **Import App Package**:

Reviewer: ```suggestion
1. Select **Import App Package…** from the **File** menu:
```

    {{< figure src="/attachments/refguide/modeling/import-and-export/file-import-app-project.png" >}}

2. Select your Mendix app package file and click **Open**. The **Import App Package** dialog box will appear:

    {{< figure src="/attachments/refguide/modeling/import-and-export/import-app-package.png" >}}

    You have the following options in this dialog box:

>     * Decide whether to store your app in a **New Mendix Team Server** (recommended), an **Existing Mendix Team Server**, or **Locally on disk**

Reviewer: The image also has **Private server** should we add that to the list?

Author: Good catch! Added to the list.

To learn more about importing content from the Marketplace, see [How to Use Marketplace Content in Studio Pro](/appstore/general/app-store-content/).

## 3 Exporting Various Elements

### 3.1 Exporting App Packages

To export an app, follow these steps:

> 1. Click the **File** menu in  and select **Export App Package**.

Reviewer: ```suggestion
1. Select **Export App Package…** from the **File** menu:
```

    {{< figure src="/attachments/refguide/modeling/import-and-export/file-export-package.png" >}}

> 2. In the **Export App Package** dialog box, which enables you to select the type of the package, set the export location, and select data to export (if required):

Reviewer: ```suggestion
2. In the **Export App Package** dialog box, select the **Package Type**, the **Package destination**, and whether you want to export a snapshot of the data in the current development database:
```
